packages/micropip/src/micropip/_micropip.py

```diff
@@ -604,10 +604,15 @@ def _list():
         if name in packages:
             continue
 
-        version = BUILTIN_PACKAGES[name]["version"]
-        source_ = "pyodide"
-        if pkg_source != "default channel":
-            # Pyodide package loaded from a custom URL
+        if name in BUILTIN_PACKAGES:
```

Comment:
BTW, I don't understand how `BUILTIN_PACKAGES` gets updated. As far as I understand, it's a set [converted from `pyodide_js._api.packages`](https://github.com/pyodide/pyodide/blob/8e8ccffddc669aad0b982798d93f0f10f8d11a78/packages/micropip/src/micropip/_compat_in_pyodide.py#L9) but should't we do this conversion where it's used each time instead of at import time once? Or otherwise keep it as a JsProxy?

Reply:
Aww, right it's the fixed `package.json`. We should probably use more explicit names. The link between `_api.packages` / `BUILTIN_PACKAGES`  and `packages.json` is not very explicit from the name.